Sources/Sentry/SentryProfiler.mm

```diff
@@ -657,107 +751,71 @@ - (void)captureEnvelope
         @"model" : isEmulated ? sentry_getSimulatorDeviceModel() : sentry_getDeviceModel()
     };
 
-    const auto profileID = [[SentryId alloc] init];
     profile[@"profile_id"] = profileID.sentryIdString;
-    const auto profileDuration = getDurationNs(_startTimestamp, _endTimestamp);
     profile[@"duration_ns"] = [@(profileDuration) stringValue];
-    profile[@"truncation_reason"] = profilerTruncationReasonName(_truncationReason);
-    profile[@"platform"] = _transactions.firstObject.platform;
-    profile[@"environment"] = _hub.scope.environmentString ?: _hub.getClient.options.environment;
+    profile[@"truncation_reason"]
+        = profilerTruncationReasonName(_gCurrentProfiler->_truncationReason);
+    profile[@"platform"] = platform;
+    profile[@"environment"] = _gCurrentProfiler->_hub.scope.environmentString
+        ?: _gCurrentProfiler->_hub.getClient.options.environment;
     profile[@"timestamp"] = [[SentryCurrentDate date] sentry_toIso8601String];
-    profile[@"release"] = _hub.getClient.options.releaseName;
-
-    profile[@"measurements"] = metrics;
+    profile[@"release"] = _gCurrentProfiler->_hub.getClient.options.releaseName;
+}
 
-#    if SENTRY_HAS_UIKIT
-    const auto slowTimestamps
-        = processFrameRenders(_frameInfo.slowFrameTimestamps, _startTimestamp, profileDuration);
-    if (slowTimestamps.count > 0) {
-        metrics[kSentryProfilerSerializationKeySlowFrameRenders] =
-            @{ @"unit" : @"nanosecond", @"values" : slowTimestamps };
-    }
+/** @return serialize info from all transactions that occurred while profiler was running */
++ (NSDictionary *)serializeInfoForTransaction:(SentryTransaction *)transaction
```

Comment:
`l`: The comment states `serialize info from all transactions`, but we only pass in one transaction. I think the comment needs some update or we need to change the code.